processing/src/main/java/org/apache/druid/segment/AutoTypeColumnMerger.java

```diff
@@ -87,6 +88,7 @@ public class AutoTypeColumnMerger implements DimensionMergerV9
 
   public AutoTypeColumnMerger(
```

Comment:
i didn't make it nullable because still needed the fields, and this would be nice to keep if we ever want to start prefixing the name of the base table too (assuming larger changes dont also mean we are using something besides dimension mergers to write segments), but for now at least added javadocs to this and `StringDimensionMergerV9`